pandas/io/excel/_base.py

```diff
@@ -1558,25 +1548,41 @@ def parse(
         DataFrame or dict of DataFrames
             DataFrame from the passed in Excel file.
         """
+        allowed_kwargs = [
+            "header",
+            "names",
+            "index_col",
+            "usecols",
+            "squeeze",
+            "dtype",
+            "engine",
+            "converters",
+            "true_values",
+            "false_values",
+            "skiprows",
+            "nrows",
+            "na_values",
+            "keep_default_na",
+            "na_filter",
+            "verbose",
+            "parse_dates",
+            "date_parser",
+            "thousands",
+            "decimal",
+            "comment",
+            "skipfooter",
+            "convert_float",
+        ]
+        if any(key in allowed_kwargs for key in kwds):
+            warnings.warn(
+                "Specifying arguments other than sheet_name by position is deprecated."
```

Comment:
The if condition here will evaluate to true when a user passes an argument by keyword, no? We only want to warn when an argument is passed by position.